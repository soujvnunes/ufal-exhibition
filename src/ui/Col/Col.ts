import styled, { css } from "styled-components";
import { ColEdges, ColProps } from "types";
import Flex from "ui/Flex";
import { token as t } from "utils";

const getColSize = (col: number) => (100 * col) / 12;
const getCol = (col: ColEdges, media?: "md" | "lg") => css`
  ${media && t(media)} {
    margin-left: ${col.start && `${getColSize(col.start)}%`};
    margin-right: ${col.end && `${getColSize(col.end)}%`};
    flex-basis: ${col.mid && `${getColSize(col.mid)}%`};
    flex-grow: ${col.mid && 0};
    max-width: ${col.mid && `${getColSize(col.mid)}%`};
  }
`;
const Col = styled(Flex).attrs({ as: "li" })<ColProps>`
  padding-left: ${t("padding")};
  padding-top: ${t("padding")};
  ${(props) => props.$sm && getCol(props.$sm)};
  ${(props) => props.$md && getCol(props.$md, "md")};
  ${(props) => props.$lg && getCol(props.$lg, "lg")};
`;

Col.displayName = "Col";
export default Col;
