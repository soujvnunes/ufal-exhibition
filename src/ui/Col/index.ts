export { default } from "./Col";
export * from "./Col.type";
export * from "./Col.util";
