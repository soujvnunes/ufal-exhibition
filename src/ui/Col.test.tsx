import renderer from "react-test-renderer";
import Col, { getColSize, mapCols, mapMediaQueries } from "ui/Col";
import { find } from "styled-components/test-utils";
import { render } from "@testing-library/react";
import { theme, ThemeMediaQuery } from "app";

describe("Col", () => {
  it("should render without crash", () => {
    render(<Col />);

    expect(find(document.body, Col)).toBeTruthy();
  });
  it("should render default styles correctly", () => {
    const tree = renderer.create(<Col theme={theme} />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("padding-left", theme.size(8));
    expect(tree).toHaveStyleRule("padding-top", theme.size(8));
  });
  mapCols.forEach((col) => {
    it(`should render ${col} column(s) with ${col} start/end span(s)`, () => {
      const tree = renderer
        .create(<Col $start={col} $mid={col} $end={col} theme={theme} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("flex-grow", "0");
      expect(tree).toHaveStyleRule("margin-left", getColSize(col));
      expect(tree).toHaveStyleRule("margin-right", getColSize(col));
      expect(tree).toHaveStyleRule("flex-basis", getColSize(col));
      expect(tree).toHaveStyleRule("max-width", getColSize(col));
    });
    mapMediaQueries.forEach((_media) => {
      const media = _media as ThemeMediaQuery;
      const isDefault = media === "sm";
      const options = isDefault
        ? undefined
        : {
            media: {
              md: "(min-width:640px)",
              lg: "(min-width:1280px)",
            }[media],
          };

      it(`should render ${col} column(s) with ${col} start/end span(s) on ${media} media`, () => {
        const tree = renderer
          .create(
            <Col
              $start={{ [media]: col }}
              $mid={{ [media]: col }}
              $end={{ [media]: col }}
              theme={theme}
            />,
          )
          .toJSON();

        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule(
          "flex-grow",
          !isDefault ? undefined : "0",
          options,
        );
        expect(tree).toHaveStyleRule("margin-left", getColSize(col), options);
        expect(tree).toHaveStyleRule("margin-right", getColSize(col), options);
        expect(tree).toHaveStyleRule("flex-basis", getColSize(col), options);
        expect(tree).toHaveStyleRule("max-width", getColSize(col), options);
      });
    });
  });
});
