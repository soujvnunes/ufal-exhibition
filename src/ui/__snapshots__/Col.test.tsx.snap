// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should render 1 column(s) with 1 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 8.333333333333334%;
  -webkit-flex-basis: 8.333333333333334%;
  -ms-flex-preferred-size: 8.333333333333334%;
  flex-basis: 8.333333333333334%;
  max-width: 8.333333333333334%;
  margin-right: 8.333333333333334%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 1 column(s) with 1 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 8.333333333333334%;
  -webkit-flex-basis: 8.333333333333334%;
  -ms-flex-preferred-size: 8.333333333333334%;
  flex-basis: 8.333333333333334%;
  max-width: 8.333333333333334%;
  margin-right: 8.333333333333334%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 1 column(s) with 1 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 8.333333333333334%;
    -webkit-flex-basis: 8.333333333333334%;
    -ms-flex-preferred-size: 8.333333333333334%;
    flex-basis: 8.333333333333334%;
    max-width: 8.333333333333334%;
    margin-right: 8.333333333333334%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 1 column(s) with 1 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 8.333333333333334%;
    -webkit-flex-basis: 8.333333333333334%;
    -ms-flex-preferred-size: 8.333333333333334%;
    flex-basis: 8.333333333333334%;
    max-width: 8.333333333333334%;
    margin-right: 8.333333333333334%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 2 column(s) with 2 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 16.666666666666668%;
  -webkit-flex-basis: 16.666666666666668%;
  -ms-flex-preferred-size: 16.666666666666668%;
  flex-basis: 16.666666666666668%;
  max-width: 16.666666666666668%;
  margin-right: 16.666666666666668%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 2 column(s) with 2 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 16.666666666666668%;
  -webkit-flex-basis: 16.666666666666668%;
  -ms-flex-preferred-size: 16.666666666666668%;
  flex-basis: 16.666666666666668%;
  max-width: 16.666666666666668%;
  margin-right: 16.666666666666668%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 2 column(s) with 2 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 16.666666666666668%;
    -webkit-flex-basis: 16.666666666666668%;
    -ms-flex-preferred-size: 16.666666666666668%;
    flex-basis: 16.666666666666668%;
    max-width: 16.666666666666668%;
    margin-right: 16.666666666666668%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 2 column(s) with 2 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 16.666666666666668%;
    -webkit-flex-basis: 16.666666666666668%;
    -ms-flex-preferred-size: 16.666666666666668%;
    flex-basis: 16.666666666666668%;
    max-width: 16.666666666666668%;
    margin-right: 16.666666666666668%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 3 column(s) with 3 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 25%;
  -webkit-flex-basis: 25%;
  -ms-flex-preferred-size: 25%;
  flex-basis: 25%;
  max-width: 25%;
  margin-right: 25%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 3 column(s) with 3 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 25%;
  -webkit-flex-basis: 25%;
  -ms-flex-preferred-size: 25%;
  flex-basis: 25%;
  max-width: 25%;
  margin-right: 25%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 3 column(s) with 3 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 25%;
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    max-width: 25%;
    margin-right: 25%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 3 column(s) with 3 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 25%;
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    max-width: 25%;
    margin-right: 25%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 4 column(s) with 4 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 33.333333333333336%;
  -webkit-flex-basis: 33.333333333333336%;
  -ms-flex-preferred-size: 33.333333333333336%;
  flex-basis: 33.333333333333336%;
  max-width: 33.333333333333336%;
  margin-right: 33.333333333333336%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 4 column(s) with 4 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 33.333333333333336%;
  -webkit-flex-basis: 33.333333333333336%;
  -ms-flex-preferred-size: 33.333333333333336%;
  flex-basis: 33.333333333333336%;
  max-width: 33.333333333333336%;
  margin-right: 33.333333333333336%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 4 column(s) with 4 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 33.333333333333336%;
    -webkit-flex-basis: 33.333333333333336%;
    -ms-flex-preferred-size: 33.333333333333336%;
    flex-basis: 33.333333333333336%;
    max-width: 33.333333333333336%;
    margin-right: 33.333333333333336%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 4 column(s) with 4 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 33.333333333333336%;
    -webkit-flex-basis: 33.333333333333336%;
    -ms-flex-preferred-size: 33.333333333333336%;
    flex-basis: 33.333333333333336%;
    max-width: 33.333333333333336%;
    margin-right: 33.333333333333336%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 5 column(s) with 5 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 41.666666666666664%;
  -webkit-flex-basis: 41.666666666666664%;
  -ms-flex-preferred-size: 41.666666666666664%;
  flex-basis: 41.666666666666664%;
  max-width: 41.666666666666664%;
  margin-right: 41.666666666666664%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 5 column(s) with 5 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 41.666666666666664%;
  -webkit-flex-basis: 41.666666666666664%;
  -ms-flex-preferred-size: 41.666666666666664%;
  flex-basis: 41.666666666666664%;
  max-width: 41.666666666666664%;
  margin-right: 41.666666666666664%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 5 column(s) with 5 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 41.666666666666664%;
    -webkit-flex-basis: 41.666666666666664%;
    -ms-flex-preferred-size: 41.666666666666664%;
    flex-basis: 41.666666666666664%;
    max-width: 41.666666666666664%;
    margin-right: 41.666666666666664%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 5 column(s) with 5 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 41.666666666666664%;
    -webkit-flex-basis: 41.666666666666664%;
    -ms-flex-preferred-size: 41.666666666666664%;
    flex-basis: 41.666666666666664%;
    max-width: 41.666666666666664%;
    margin-right: 41.666666666666664%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 6 column(s) with 6 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 50%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  max-width: 50%;
  margin-right: 50%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 6 column(s) with 6 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 50%;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  max-width: 50%;
  margin-right: 50%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 6 column(s) with 6 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 50%;
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    margin-right: 50%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 6 column(s) with 6 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 50%;
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    margin-right: 50%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 7 column(s) with 7 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 58.333333333333336%;
  -webkit-flex-basis: 58.333333333333336%;
  -ms-flex-preferred-size: 58.333333333333336%;
  flex-basis: 58.333333333333336%;
  max-width: 58.333333333333336%;
  margin-right: 58.333333333333336%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 7 column(s) with 7 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 58.333333333333336%;
  -webkit-flex-basis: 58.333333333333336%;
  -ms-flex-preferred-size: 58.333333333333336%;
  flex-basis: 58.333333333333336%;
  max-width: 58.333333333333336%;
  margin-right: 58.333333333333336%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 7 column(s) with 7 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 58.333333333333336%;
    -webkit-flex-basis: 58.333333333333336%;
    -ms-flex-preferred-size: 58.333333333333336%;
    flex-basis: 58.333333333333336%;
    max-width: 58.333333333333336%;
    margin-right: 58.333333333333336%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 7 column(s) with 7 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 58.333333333333336%;
    -webkit-flex-basis: 58.333333333333336%;
    -ms-flex-preferred-size: 58.333333333333336%;
    flex-basis: 58.333333333333336%;
    max-width: 58.333333333333336%;
    margin-right: 58.333333333333336%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 8 column(s) with 8 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 66.66666666666667%;
  -webkit-flex-basis: 66.66666666666667%;
  -ms-flex-preferred-size: 66.66666666666667%;
  flex-basis: 66.66666666666667%;
  max-width: 66.66666666666667%;
  margin-right: 66.66666666666667%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 8 column(s) with 8 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 66.66666666666667%;
  -webkit-flex-basis: 66.66666666666667%;
  -ms-flex-preferred-size: 66.66666666666667%;
  flex-basis: 66.66666666666667%;
  max-width: 66.66666666666667%;
  margin-right: 66.66666666666667%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 8 column(s) with 8 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 66.66666666666667%;
    -webkit-flex-basis: 66.66666666666667%;
    -ms-flex-preferred-size: 66.66666666666667%;
    flex-basis: 66.66666666666667%;
    max-width: 66.66666666666667%;
    margin-right: 66.66666666666667%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 8 column(s) with 8 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 66.66666666666667%;
    -webkit-flex-basis: 66.66666666666667%;
    -ms-flex-preferred-size: 66.66666666666667%;
    flex-basis: 66.66666666666667%;
    max-width: 66.66666666666667%;
    margin-right: 66.66666666666667%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 9 column(s) with 9 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 75%;
  -webkit-flex-basis: 75%;
  -ms-flex-preferred-size: 75%;
  flex-basis: 75%;
  max-width: 75%;
  margin-right: 75%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 9 column(s) with 9 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 75%;
  -webkit-flex-basis: 75%;
  -ms-flex-preferred-size: 75%;
  flex-basis: 75%;
  max-width: 75%;
  margin-right: 75%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 9 column(s) with 9 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 75%;
    -webkit-flex-basis: 75%;
    -ms-flex-preferred-size: 75%;
    flex-basis: 75%;
    max-width: 75%;
    margin-right: 75%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 9 column(s) with 9 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 75%;
    -webkit-flex-basis: 75%;
    -ms-flex-preferred-size: 75%;
    flex-basis: 75%;
    max-width: 75%;
    margin-right: 75%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 10 column(s) with 10 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 83.33333333333333%;
  -webkit-flex-basis: 83.33333333333333%;
  -ms-flex-preferred-size: 83.33333333333333%;
  flex-basis: 83.33333333333333%;
  max-width: 83.33333333333333%;
  margin-right: 83.33333333333333%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 10 column(s) with 10 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 83.33333333333333%;
  -webkit-flex-basis: 83.33333333333333%;
  -ms-flex-preferred-size: 83.33333333333333%;
  flex-basis: 83.33333333333333%;
  max-width: 83.33333333333333%;
  margin-right: 83.33333333333333%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 10 column(s) with 10 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 83.33333333333333%;
    -webkit-flex-basis: 83.33333333333333%;
    -ms-flex-preferred-size: 83.33333333333333%;
    flex-basis: 83.33333333333333%;
    max-width: 83.33333333333333%;
    margin-right: 83.33333333333333%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 10 column(s) with 10 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 83.33333333333333%;
    -webkit-flex-basis: 83.33333333333333%;
    -ms-flex-preferred-size: 83.33333333333333%;
    flex-basis: 83.33333333333333%;
    max-width: 83.33333333333333%;
    margin-right: 83.33333333333333%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 11 column(s) with 11 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 91.66666666666667%;
  -webkit-flex-basis: 91.66666666666667%;
  -ms-flex-preferred-size: 91.66666666666667%;
  flex-basis: 91.66666666666667%;
  max-width: 91.66666666666667%;
  margin-right: 91.66666666666667%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 11 column(s) with 11 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 91.66666666666667%;
  -webkit-flex-basis: 91.66666666666667%;
  -ms-flex-preferred-size: 91.66666666666667%;
  flex-basis: 91.66666666666667%;
  max-width: 91.66666666666667%;
  margin-right: 91.66666666666667%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 11 column(s) with 11 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 91.66666666666667%;
    -webkit-flex-basis: 91.66666666666667%;
    -ms-flex-preferred-size: 91.66666666666667%;
    flex-basis: 91.66666666666667%;
    max-width: 91.66666666666667%;
    margin-right: 91.66666666666667%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 11 column(s) with 11 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 91.66666666666667%;
    -webkit-flex-basis: 91.66666666666667%;
    -ms-flex-preferred-size: 91.66666666666667%;
    flex-basis: 91.66666666666667%;
    max-width: 91.66666666666667%;
    margin-right: 91.66666666666667%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 12 column(s) with 12 start/end span(s) 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 100%;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  margin-right: 100%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 12 column(s) with 12 start/end span(s) on sm media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-left: 100%;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
  margin-right: 100%;
}

<li
  className="c0"
/>
`;

exports[`Col should render 12 column(s) with 12 start/end span(s) on lg media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:80rem) {
  .c0 {
    margin-left: 100%;
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    margin-right: 100%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render 12 column(s) with 12 start/end span(s) on md media 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

@media (min-width:40rem) {
  .c0 {
    margin-left: 100%;
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    margin-right: 100%;
  }
}

<li
  className="c0"
/>
`;

exports[`Col should render default styles correctly 1`] = `
.c0 {
  padding-left: var(--grid-padding);
  padding-top: var(--grid-padding);
}

<li
  className="c0"
/>
`;
